src/com/google/enterprise/adaptor/Config.java

```diff
@@ -643,20 +643,40 @@ long getAdaptorIncrementalPollPeriodMillis() {
     return Long.parseLong(getValue("adaptor.incrementalPollPeriodSecs")) * 1000;
   }
 
+  public long validateTimeout(String property, String defaultValue,
+      boolean allowNeg, boolean allowEmpty) {
+    String secondsAsString = getValue(property).trim();
+    try {
+      if ((null == secondsAsString) || "0".equals(secondsAsString) ||
+          (("".equals(secondsAsString) && !(allowEmpty))) ||
+          (secondsAsString.startsWith("-") && !(allowNeg))) {
+        log.log(Level.WARNING, "Invalid value for " + property
+                + ", it has been set to default " + defaultValue + " seconds.");
+      } else {
+          return Long.parseLong(secondsAsString) * 1000;
+      }
+    } catch (NumberFormatException e) {
+      log.log(Level.WARNING, "Invalid value for " + property
+              + ", it has been set to default " + defaultValue + " seconds.");
+    }
+    // For invalid value, returning default
+    // We do not reach this line if value is valid
+    return Long.parseLong(defaultValue) * 1000;
+  }
+
   long getAdaptorDocHeaderTimeoutMillis() {
-    return Long.parseLong(getValue("adaptor.docHeaderTimeoutSecs")) * 1000;
+    return validateTimeout("adaptor.docHeaderTimeoutSecs", "30", false, false);
```

Comment:
You are restating the default values supplied in the constructor here. I don't like that for 2 reasons:  
1) If the defaults are changed in the future, the change may be missed here.  
2) The default value may have been overridden by the connector using overrideKey(). (The FS adaptor overrides these values)

One solution might be to pull the default values from defaultConfig, but you need to be wary of nulls and computed values (not the case for these).